src/inapp/inapp.ts

```diff
@@ -153,25 +162,35 @@ export function getInAppMessages(
          */
         if (!inAppMessage.content) {
           const cachedMessage = cachedMessages.find(
-            ([messageId, _]) => inAppMessage.messageId === messageId
+            ([messageId]) => inAppMessage.messageId === messageId
           );
           if (cachedMessage) allMessages.push(cachedMessage[1]);
         } else {
           allMessages.push(inAppMessage);
-          /** add NEW messages from delta response to cache */
-          if (inAppMessage.messageId) set(inAppMessage.messageId, inAppMessage);
+          if (inAppMessage.messageId)
+            newMessages.push({
+              messageId: inAppMessage.messageId,
+              message: inAppMessage as InAppMessage
+            });
         }
       });
 
+      /** add new messages to the cache if they fit in the cache */
+      const remainingQuota = await determineRemainingStorageQuota();
+      await addNewMessagesToCache(newMessages, remainingQuota);
```

Comment:
Do we use `remainingQuota` anywhere else? Any reason we can't call that util from within `addNewMessagesToCache` instead of passing it in?